core/utils/files.go

```diff
@@ -44,13 +44,13 @@ func EnsureDirAndMaxPerms(path string, perms os.FileMode) error {
 		return err
 	} else if os.IsNotExist(err) {
 		// Dir doesn't exist, create it with desired perms
-		return os.MkdirAll(path, perms|os.ModeDir)
+		return os.MkdirAll(path, perms)
```

Comment:
The `os.ModeDir` flag was creating unusable directories.

Reply:
Is this going to make our permissions too open again?